src/main/java/com/maxmind/db/Reader.java

```diff
@@ -190,7 +198,61 @@ record = this.readNode(buffer, record, bit);
         return new DatabaseRecord<>(dataRecord, ipAddress, pl);
     }
 
-    private BufferHolder getBufferHolder() throws ClosedDatabaseException {
+    /**
+     * Creates a Networks iterator and skips aliased networks.
+     * Please note that a MaxMind DB may map IPv4 networks into several locations
+     * in an IPv6 database. This iterator will iterate over all of these locations
+     * separately. To disable skipAliasedNetworks, which iterates over the IPv4 networks 
+     * only once, use the SkipAliasedNetworks parameter.
+     * 
+     * @param <T> Represents the data type(e.g., Map, HastMap, etc.).
+     * @param typeParameterClass The type of data returned by the iterator.
+     * @return Networks The Networks iterator.
+     * @throws InvalidNetworkException Exception for using an IPv6 network in ipv4-only database.
+     * @throws ClosedDatabaseException Exception for a closed databased.
+     * @throws InvalidDatabaseException Exception for an invalid database.
+     */
+    public <T> Networks<T> networks(Class<T> typeParameterClass) throws 
+        InvalidNetworkException, ClosedDatabaseException, InvalidDatabaseException {
+        return this.networks(true, typeParameterClass);
+    }
+
+    /**
+     * Creates a Networks iterator.
+     * Please note that a MaxMind DB may map IPv4 networks into several locations
+     * in an IPv6 database. This iterator will iterate over all of these locations
+     * separately. To set the iteration over the IPv4 networks once, use the
+     * SkipAliasedNetworks option.
+     * 
+     * @param <T> Represents the data type(e.g., Map, HastMap, etc.).
+     * @param skipAliasedNetworks Enable or disable skipping aliased networks.
+     * @return Networks The Networks iterator.
+     * @throws InvalidNetworkException Exception for using an IPv6 network in ipv4-only database.
+     * @throws ClosedDatabaseException Exception for a closed databased.
+     * @throws InvalidDatabaseException Exception for an invalid database.
+     */
+    public <T> Networks<T> networks(
+            boolean skipAliasedNetworks,
+            Class<T> typeParameterClass) throws
+        InvalidNetworkException, ClosedDatabaseException, InvalidDatabaseException {
+        try {
+            InetAddress ipv4 = InetAddress.getByAddress(new byte[4]);
+            InetAddress ipv6 = InetAddress.getByAddress(new byte[16]);
+            Network ipAllV4 = new Network(ipv4, 0); // Mask 32.
+            Network ipAllV6 = new Network(ipv6, 0); // Mask 128.
```

Comment:
It would save allocations to only create these when you actually need them (i.e., move the v6 stuff to the if block and the v4 stuff after the if block).